### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by Daniduran-dev**
### Milestone 3: Game logic
- Game logic finished
- Create 2 classes for the board and players
- Game working
- Use hashes to improve readability and user experience.

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    puts "a|#{@board[:a1]}|#{@board[:a2]}|#{@board[:a3]}|"
+    puts "b|#{@board[:b1]}|#{@board[:b2]}|#{@board[:b3]}|"
+    puts "c|#{@board[:c1]}|#{@board[:c2]}|#{@board[:c3]}|"
+  end
+
+  def empty_cells
+    freecells = @board.select { |_k, v| v == ' ' }
+    print freecells.keys
+  end
+
```
Comment (adaorachi) [inline lib/board.rb]: - [x] Your `main.rb` file should be responsible for all user interactions. Kindly ensure that there are no `puts/print` and `get.chomp` outside `main.rb` file. If you must output some strings, you can assign it to variables and return them and then, call them appropriately in the `main.rb` file. Here is a [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from pathwright for clarification.

**Review by adaorachi — CHANGES_REQUESTED** [review]
### Hi @Daniduran-dev :wave: 
You've done an awesome job on this project and I commend that. Well done!
However, there are a couple of changes to be implemented in order to get your project approved.

---

### STATUS: CHANGES REQUIRED :recycle: 

---

- [x] Kindly improve on your pull-request summary and title. The summary should be very descriptive, you should be able to describe what you did in this milestone and the features you implemented. Here is a [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_prs.png) for reference.

- [x] Kindly ensure you improve your commit messages and commit as often as possible as well. They should be descriptive and explicitly explains the features or units of work you have done. Currently, you only have two commit messages describing the features you implemented. 

- [x] Ensure that you do not create one class for all your logic. Your project should use at least two classes (usually it uses three: Game, Board, and Player). And also ensure that your code is logically divided amongst these classes. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/classes.png) from Pathwright)

- [**optional**] It will be nice if you cover the edge cases of verifying that a player actually inputs their names and not just an empty string before continuing the game. Your game does not break but it will help improve the user-interface.
- [**optional**] It will also be nice if you inform a player when they input a wrong key combination that does not match any of the cells.

---
### After implementing the mandatory changes required, you can request for another code review.
#### All requirements not marked with OPTIONAL are mandatory.
---

Should you have any question, kindly reach out to me on Slack with the username @adaorachi

Happy Coding :relaxed: 

MaryAnn Chukwuka.

**Comment by Daniduran-dev** [comment]
### Changes:
- Divide the code into 2 classes
- Create more methods
- Move the puts and prints to the main.rb

File: lib/board.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    wins = {}
+    @win_combo.each do |k|
+      case @board.values_at(*k)
+      when %w[X X X] then wins[k] = 'X'
+      when %w[O O O] then wins[k] = 'O'
+      end
+    end
+    wins
+  end
+end
```
Comment (bertil291utn) [inline lib/board.rb]: ## Keep a single responsibility approach 
- Create another function/method to return `@win_combo` variable
- Receive @wincombo as a parameter 
Example:

```ruby
 def winner_checking(win_combo)

```

**Review by bertil291utn — CHANGES_REQUESTED** [review]
# **Status:** Changes requested 🟠 
<p align="center"> image <br><strong>Not yet</strong> </p>

# Hey guys, you've been doing a great job, and also I hope you enjoyed coding.

### Your work is amazing you master it GitHub Flow you have a professional README file. 

However, in order to accomplish all the requirements, you required to do these changes, listed below:

- [x] Edit your pull request message  following this [guidelinede](https://cdn.filestackcontent.com/osvFJEkQHWQsgDe0PJoQ)
- When you got a not valid input should inform the player about it and asks for their move again
 - [x] Add a message suggesting it.

NICE TO HAVE [OPTIONAL] 
- It'd be great to select a number position(1,2,3,4) instead (a1, a2, b3)
- Remove this message "The winning combination is: [[:a3, :b3, :c3]]"  is quite confusing 
### Summary

-  Don't forget to work with inline comments below this comment box.

***

Guys work in the requested changes above and request another review.
Good luck. 

> “Never stop, never conform until what is good becomes even better, and what’s better becomes excellent.”

**Review by expjazz — APPROVED** [review]
## Status: Project Approved :clap: 

You did a very good job on this project, and on the changes previously requested by my colleagues :1st_place_medal: 

Good Luck and Happy Coding :pray:

_P.S. If you have any questions about my review or need help, you can message me on slack (@ Expedito Andrade).:smile:_

**Pull request merged**